setup.py

```diff
@@ -32,11 +33,19 @@ def load_requirements(file_list: Optional[Union[str, List[str]]] = None) -> List
         file_list = [file_list]
     requirements: List[str] = []
     for file in file_list:
+        path = Path(file)
+        file_dir = path.parent.resolve()
         with open(file, encoding="utf-8") as f:
-            requirements.extend(f.readlines())
-    requirements = [
-        req for req in requirements if not req.startswith("#") and not req.startswith("-")
-    ]
+            raw = f.read().splitlines()
+            requirements.extend([r for r in raw if not r.startswith("#") and not r.startswith("-")])
+            recursive_reqs = [r for r in raw if r.startswith("-r")]
+            if recursive_reqs:
```

Comment:
Small nit: the code iterates over `recursive_reqs`, so if `recursive_reqs` is empty all the code within this `if` block is a noop anyway. In such cases I think it's cleaner and more readable to not include the `if` statement.

Reply:
I actually did this for readability because I only instantiate the `filenames = []` if there are any recursive ones to look through.

Reply:
But I could just have it call it one at a time.